### ISSUE TITLE
Mattermost sets cookie rl_user_id for parent domain

### ISSUE THREAD

**Issue by JsBergbau** [mattermost/mattermost#28085]
Before filing a bug report, please check the following:

1. Confirm you’re filing a new issue. [Search existing tickets in Jira](https://mattermost.atlassian.net/jira/software/c/projects/MM/issues/) and [existing issues in GitHub](https://github.com/mattermost/mattermost/issues) to ensure that the ticket does not already exist.
2. Confirm your issue does not involve security. Please submit security issues via our [Responsible Disclosure Policy](https://mattermost.com/security-vulnerability-report/).
3. Confirm your issue is not a feature request. Please submit feature requests on [UserVoice](https://mattermost.uservoice.com/forums/306457-general).
4. Confirm your issue is not a troubleshooting question. Please submit troubleshooting questions on the [forum](https://forum.mattermost.com/).
5. File a bug report using the format below. Mattermost will confirm steps to reproduce and file in Jira, or ask for more details if there is trouble reproducing it. If there's already an existing bug in Jira, it will be linked back to the GitHub issue so you can track when it gets fixed.

#### Summary
Mattermost sets cookie rl_user_id for parent domain

#### Steps to reproduce
Login into mattermost, see cookies. You will now see a cookie rl_user_id for mattermost.example.org AND .example.org
#### Operating System
What Operating System do you use?
Win 64 Chrome and Firefox to reproduce the problem
Ubuntu 24 LTS Server for Server

#### Mattermost Server Version
Which version of the Mattermost Server did this occur on? (You can find your Mattermost Server version via **Mattermost Menu** > **About Mattermost**)

Mattermost-Version: 9.11.1
Datenbankschema-Version: 124
Build-Nummer: 10419911144
Datenbank: postgres

#### Expected behavior
Cookies are only set for mattermost.example.org

#### Observed behavior
Cookie is also set for .example.org
This leads to problems, because all websites under example.org get the rl_user_id cookie set.

#### Log Output
Output from the log files if applicable.

#### Additional Information
Additional helpful information, such as error messages and/or screenshots.

#### Possible fixes
If you can, link to the line of code that might be responsible for the problem.

**Comment by amyblais** [comment]
Opened https://mattermost.atlassian.net/browse/MM-60309.

**Comment by agnivade** [comment]
@JsBergbau - could you confirm if you are using focalboard? Is the cookie being set by Mattermost or focalboard?

**Comment by JsBergbau** [comment]
I don't use focalboard. Cookie is set by mattermost.
I found a way to avoid these cookies. You have to disable Telemtetry like described here https://docs.mattermost.com/manage/telemetry.html#id2 AND you have to restart mattermost service, only then none of the rl_user_id Cookies are set.

**Comment by agnivade** [comment]
Yes, if you disable telemetry then it won't be set. But even if you do, it shouldn't set for the parent domain. Could you confirm if you have the siteURL set and what is the value of that? Also, could you please post a screenshot of the cookies tab of your browser? If it's sensitive information, please feel free to DM me on community.mattermost.com @agniva.de.sarker.

**Comment by JsBergbau** [comment]
Thanks for digging into this.

> Yes, if you disable telemetry then it won't be set. 

And you have to restart mattermost-Service. That was the missing part.

> Could you confirm if you have the siteURL set and what is the value of that? 

Yes, site URL is set to mattermost.example.org

> Also, could you please post a screenshot of the cookies tab of your browser? 

There must have been a change. Now cookies aren't set for parent-domain anymore, as it should be. All cookies are now set for .mattermost.example.org

![Image](https://github.com/user-attachments/assets/9a5992b9-8912-4e45-8d80-c015fa0d926b)

**Comment by agnivade** [comment]
Alright. It must have been some temporary issue then. Going ahead and closing this out.

**Comment by JsBergbau** [comment]
I wouldn't say that temporary, because I checked in November or December again and there still the cookie was set for parent domain then.
